### ISSUE TITLE
PR4-03: Add smoke test for one coarse training step with memory enabled

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#461]
# PR4-03: Add smoke test for one coarse training step with memory enabled

## Goal

Add a small smoke test proving that the wired criterion can run through at least two training-style calls with the real `TraceMemoryConsistencyCriterion` and PR1 batch identity fields. This catches shape, device, and output-contract regressions before running full site54 jobs.

This is not an end-to-end SEG-Y test. It should be CPU-only and synthetic.

## Target files

- Prefer adding to:
  - `packages/seisai-engine/tests/test_fbpick_coarse_train_trace_memory_wiring.py`
- Or extend:
  - `packages/seisai-engine/tests/test_fbpick_coarse_trace_memory_consistency.py`

## Required synthetic batch

Use a tiny batch with the required fields:

```python
batch = {
    'file_index': torch.tensor([0], dtype=torch.long),
    'primary_key_id': torch.tensor([0], dtype=torch.long),
    'secondary_key_id': torch.tensor([0], dtype=torch.long),
    'anchor_raw_indices_view': torch.tensor([[10, 11, 12]], dtype=torch.long),
    'meta': {
        'factor': torch.tensor([1.0]),
        'start': torch.tensor([0.0]),
        'trace_tshift_view': torch.zeros((1, 3), dtype=torch.float32),
        'trace_valid': torch.ones((1, 3), dtype=torch.bool),
        'fb_idx_view': torch.full((1, 3), -1, dtype=torch.long),
        'dt_eff_sec': torch.tensor([0.001], dtype=torch.float32),
    },
}
```

Use a fake base criterion that is differentiable with respect to `pred`:

```python
def base_criterion(pred, target, batch):
    logits = get_pick_logits(pred)
    return logits.sum() * 0.0 + logits.new_tensor(2.0)
```

Use `TraceMemoryConsistencyCriterion` through the same wrapper helper that PR4 wires into `train.py`:

```python
criterion = maybe_wrap_trace_memory_consistency(
    base_criterion=base_criterion,
    cfg=enabled_cfg,
    fb_sigma_ms=2.0,
)
```

## Required assertions

1. First call has empty memory and returns the base loss only:

```python
loss1 = criterion(logits1, target, batch)
assert loss1.item() == pytest.approx(2.0)
assert criterion.memory_bank.stored_slot_count() == 3
```

2. Second call uses previous memory and returns a finite loss greater than base loss when `weight > 0` and `warmup_batches == 0`:

```python
loss2 = criterion(logits2, target, batch)
assert torch.isfinite(loss2)
assert loss2.item() > 2.0
```

3. Backward works:

```python
loss2.backward()
assert logits2.grad is not None
```

4. Warmup behavior remains correct:

With `warmup_batches=2`, the second call should have `trace_memory/used_rows > 0` but effective weight `0.0`, so total loss remains the base loss.

## Acceptance criteria

- The test exercises `maybe_wrap_trace_memory_consistency()`, not only direct class construction.
- The test uses the PR1 batch identity fields: `file_index`, `primary_key_id`, `secondary_key_id`, `anchor_raw_indices_view`.
- The test is CPU-only and runs in under a few seconds.
- The test does not require real SEG-Y, `segyio`, or dataset construction.
